src/main/java/io/vertx/core/impl/ContextImpl.java

```diff
@@ -279,5 +279,11 @@ private void setTCCL() {
     Thread.currentThread().setContextClassLoader(tccl);
   }
 
-
+  public int getInstanceCount(){
+    if(deployment == null || deployment.deploymentOptions() == null) {
+      // implicitly there is the current instance of the verticle running always
```

Comment:
Gotcha, so should the logic then be

if(deployment == null) {
   return 0;
} else if ( deployment.deploymentOptions == null) {
  return 1;
}